spyder/widgets/editor.py

```diff
@@ -1558,6 +1583,11 @@ def __check_file_status(self, index):
         # Finally, resetting temporary flag:
         self.__file_status_flag = False
 
+    def __modify_stack_title(self):
+        for index, finfo in enumerate(self.data):
+            state = finfo.editor.document().isModified()
+            self.set_stack_title(index, state)

```

Comment:
Then this piece of code could also solve issue #2907.

Could you verify if that's the code or not?